### PR TITLE
Clear some clang warnings

### PR DESCRIPTION
**Pull request by paulmelnikow**

### CODE REVIEW

File: src/mongoc/mongoc-counters.c [hunk 1]
```diff
@@ -23,6 +23,7 @@
 #include <string.h>
 
 #ifdef BSON_OS_UNIX
+#undef MAX
```
Comment (hanumantmk) [inline src/mongoc/mongoc-counters.c]: the ordering around this seems somewhat platform specific (sys/mman.h either doesn't pull MAX for me on linux, or has already been included).

I'm going to drop this change out and try to pull something together to get rid of all the non-prefixed MACROS, which as you've noted is the more robust approach.

**Comment by hanumantmk** [comment]
I'll take care of MAX, MIN, RETURN, etc... in something more holistic.

Thanks for the contribution!

File: src/mongoc/mongoc-counters.c (hunk 1 above)
Comment (paulmelnikow) [inline src/mongoc/mongoc-counters.c]: Sounds good.

**Comment by paulmelnikow** [comment]
Thanks!

**Pull request closed without merging**